src/components/service-metrics/controllers.tsx

```diff
@@ -44,69 +44,98 @@ import {
+  readonly hour?: number;
+  readonly minute?: number;
+  readonly month?: number;
+  readonly second?: number;
+  readonly year?: number;
+};
+
+type Range = RangeObject | string;
+
 // check if user-provided string is a number
-export function isNumeric(value: any) {
+export function isNumeric(value: string): boolean {
   return /^\d+$/.test(value);
 }
 
-export function sanitiseMomentInput(date: moment.MomentInputObject): moment.MomentInputObject {
-  return {
-    // if users enter something other than a number, return current date-time value
-    day: isNumeric(date.day) ? date.day : moment().date(),
-    hour: isNumeric(date.hour) ? date.hour : moment().hour(),
-    minute: isNumeric(date.minute) ? date.minute : moment().minute(),
-    month: isNumeric(date.month) ? date.month - 1 : moment().month(),
-    year: isNumeric(date.year) ? date.year : moment().year(),
-  };
+function isValidDateObject(obj: RangeObject): boolean {
+  if (obj.month && (obj.month < 1 || obj.month > 12)) {
+    return false;
+  }
+
+  if (obj.day && (obj.day < 1 || obj.day > 31)) {
+    return false;
+  }
+
+  if (obj.hour && (obj.hour < 0 || obj.hour > 23)) {
+    return false;
+  }
+
+  if (obj.minute && (obj.minute < 0 || obj.minute > 59)) {
+    return false;
+  }
+
+  if (obj.second && (obj.second < 0 || obj.second > 59)) {
+    return false;
+  }
+
+  return true;
 }
 
-export function parseRange(start: string | moment.MomentInputObject, stop: string | moment.MomentInputObject): IRange {
-  let rangeStart = typeof start === 'object' ? moment(sanitiseMomentInput(start)).isValid() ? 
-    moment(sanitiseMomentInput(start)) : moment() : moment(start);
-  let rangeStop = typeof stop === 'object' ? moment(sanitiseMomentInput(stop)).isValid() ? 
-    moment(sanitiseMomentInput(stop)) : moment() : moment(stop);
+export function objectToDate(obj: RangeObject): Date {
+  const now = new Date();
 
-  if (
-    rangeStart.isBefore(
-      rangeStop
-        .clone()
-        .subtract(1, 'year')
-        .subtract(1, 'week'),
-    )
-  ) {
-    throw new UserFriendlyError(
-      'Invalid time range provided. Cannot handle more than a year of metrics',
-    );
+  if (!isValidDateObject(obj)) {
+    return now;
   }
 
-  if (
-    rangeStop.isBefore(
-      moment()
-        .subtract(1, 'years')
-        .subtract(1, 'week'),
-    ) ||
-    rangeStart.isBefore(
-      moment()
-        .subtract(1, 'years')
-        .subtract(1, 'week'),
-    )
-  ) {
-    throw new UserFriendlyError(
-      'Invalid time range provided. Cannot handle over a year old metrics',
-    );
+  return new Date(
+    obj.year || now.getFullYear(),
+    obj.month ? obj.month - 1 : now.getMonth(),
+    obj.day || now.getDate(),
+    obj.hour !== undefined ? obj.hour : now.getHours(),
+    obj.minute !== undefined ? obj.minute : now.getMinutes(),
+    obj.second !== undefined ? obj.second : now.getSeconds(),
+  );
+}
+
+export function parseRange(start: Range, stop: Range): IRange {
```

Comment:
- to note: there is now a minor change to populating range inputs if fields are left empty
previously we populated with current time (minus X) but now we populate with 00:00
that's fine, we still handle a broken user submission

- on load we populate time close to now, whereas before we used to do minus 1 hour
 which one is expected? see below

PROD
<img width="507" alt="Screenshot 2021-06-22 at 11 53 56" src="https://user-images.githubusercontent.com/3758555/122912754-95dc0180-d350-11eb-8b03-7fc1ebf57133.png">

DEV
<img width="511" alt="Screenshot 2021-06-22 at 11 54 06" src="https://user-images.githubusercontent.com/3758555/122912860-ae4c1c00-d350-11eb-9a45-5b59ce4691e0.png">

Reply:
> on load we populate time close to now, whereas before we used to do minus 1 hour
which one is expected?

I don't know if there was ever an agreement about this if I'm honest... My thinking CURRENTLY is that the `end time` should be set to `new Date()`, where as the `start time` should be `sub(new Date(), { hours: 1 })`.

Do you have any preference?

Reply:
yep agreed, on load have `now` and `now-1h`

Reply:
when deployed it'a as per production